cohortextractor/emis_backend.py

```diff
@@ -193,7 +173,19 @@ def get_queries(self, covariate_definitions):
           {joins_str}
         WHERE {output_columns["population"]} = 1
         """
-        all_queries = []
+        patient_no_duplicates_query = f"""
+        CREATE TABLE IF NOT EXISTS {self.patient_no_duplicates_table} AS (
+            SELECT *
+            FROM {PATIENT_TABLE}
+            WHERE registration_id NOT IN (
+                SELECT registration_id
+                FROM {PATIENT_TABLE}
+                GROUP BY registration_id
+                HAVING COUNT(*) > 1
```

Comment:
Why the `HAVING` in this sub-query? I've not used SQL Server; does it not have a `SELECT DISTINCT`, for example?

Reply:
We're interested in the patients that do not have duplicates.  For instance, if we have two entries for the same `registration_id` corresponding to a 23yo man from Gillingham (Kent) and a 93yo woman from Gillingham (Dorset), we want to ignore both, since there's no way of working out whether an observation (eg a diagnosis) belongs to one of the other.

Does that make sense?

Reply:
I knew this, but in reading your (excellent) explanation I realised it'd be very good to include a version of this explanation as a comment.

Reply:
Good point.  Have added a comment.

Reply:
That makes sense and is an excellent explanation. (👍🏻 for using the place I grew up and the place my family used to go on holiday every year, too!) Thank you for adding the comment as my first thought -- as you will have guessed -- was "Don't we just want a list of registration IDs?"